torch/_decomp/decompositions.py

```diff
@@ -3282,66 +3330,71 @@ def _compute_source_index(scale, dst_index, align_corners):
-    y_f32 = y_f32.unsqueeze(-1)
-
-    x = x_f32.to(torch.int64)
-    y = y_f32.to(torch.int64)
-
-    # We are using torch.where instead of torch.clamp below due to an expected failure
-    # in test_aot_autograd_symbolic_exhaustive_nn_functional_interpolate_bilinear_cpu_float32 test
-    # torch.ops.aten.clamp.default(add, None, sub) on int64 input tensor is returning float32 and
-    # fails with torch.ops.aten._unsafe_index.Tensor(primals_1, [None, None, _to_copy_1, clamp_2])
-    # RuntimeError: _unsafe_index found unexpected index type Float
-    # xp1 = (x + 1).clamp(max=in_w - 1); yp1 = (y + 1).clamp(max=in_h - 1)
-    xp1 = torch.where(x < in_w - 1, x + 1, x)
-    yp1 = torch.where(y < in_h - 1, y + 1, y)
-
-    v1 = aten._unsafe_index(input, [None, None, y, x])
-    v2 = aten._unsafe_index(input, [None, None, y, xp1])
-    v3 = aten._unsafe_index(input, [None, None, yp1, x])
-    v4 = aten._unsafe_index(input, [None, None, yp1, xp1])
 
-    dtype = torch.float32 if not input.is_floating_point() else input.dtype
-    if not input.is_floating_point():
-        v1 = v1.to(dtype)
-        v2 = v2.to(dtype)
-        v3 = v3.to(dtype)
-        v4 = v4.to(dtype)
-
-    yscale = (y_f32 - y).clamp(0.0, 1.0).to(dtype)
-    xscale = (x_f32 - x).clamp(0.0, 1.0).to(dtype)
+    def get_values(inp, out, scales, nsqueeze):
+        # First Calculate scaling factor
+        scale_factor = _compute_scale(inp, out, align_corners, scales)
+        # We have to create arange with int64 dtype and use .to in order to avoid
+        # additional kernels creation in inductor and get a perf slowdown
+        i = torch.arange(out, device=input.device).to(dtype=dtype)
+
+        x_f32 = _compute_source_index(scale_factor, i, align_corners).clamp(min=0.0)
+        for _ in range(nsqueeze):
+            x_f32 = x_f32.unsqueeze(-1)
+        x = x_f32.to(torch.int64)
+
+        # We are using torch.where instead of torch.clamp below due to an expected failure
+        # in test_aot_autograd_symbolic_exhaustive_nn_functional_interpolate_bilinear_cpu_float32 test
+        # torch.ops.aten.clamp.default(add, None, sub) on int64 input tensor is returning float32 and
+        # fails with torch.ops.aten._unsafe_index.Tensor(primals_1, [None, None, _to_copy_1, clamp_2])
+        # RuntimeError: _unsafe_index found unexpected index type Float
+        # xp1 = (x + 1).clamp(max=in_w - 1); yp1 = (y + 1).clamp(max=in_h - 1)
+        xp1 = torch.where(x < inp - 1, x + 1, x)
+        return x_f32, x, xp1
+
+    values = [
+        get_values(inp, out, scales, d - 1 - i)
+        for i, (inp, out, scales) in enumerate(zip(inp_size, output_size, scales))
+    ]
+    xs_f32, xs, xp1s = list(zip(*values))
 
-    # x1 * (1 - alpha) + x2 * alpha == x1 + (x2 - x1) * alpha
-    q1 = v1 + torch.mul(v2 - v1, xscale)
-    q2 = v3 + torch.mul(v4 - v3, xscale)
-    result = q1 + torch.mul(q2 - q1, yscale)
+    dtype = torch.float32 if not input.is_floating_point() else input.dtype
```

Comment:
This line should be coming from my PR that I missed to remove as we already get dtype above using `utils.elementwise_dtypes`. Let's remove it here.